client/state/current-user/gravatar-status/actions.js

```diff
@@ -34,20 +41,40 @@ export function uploadGravatar( file, bearerToken, email ) {
 					dispatch( {
 						type: GRAVATAR_UPLOAD_REQUEST_SUCCESS
 					} );
+					dispatch( recordTracksEvent( 'calypso_edit_gravatar_upload_success' ) );
 				} );
 				fileReader.readAsDataURL( file );
 			} )
 			.catch( () => {
 				dispatch( {
 					type: GRAVATAR_UPLOAD_REQUEST_FAILURE
 				} );
+				dispatch( recordTracksEvent( 'calypso_edit_gravatar_upload_failure' ) );
+				dispatch( bumpStat( 'calypso_gravatar_update_error', 'unsuccessful_http_response' ) );
 			} );
 	};
 }
 
-export function receiveGravatarImageFailed( errorMessage ) {
-	return {
-		type: GRAVATAR_RECEIVE_IMAGE_FAILURE,
-		errorMessage
+export function receiveGravatarImageFailed( { errorMessage, statName } ) {
+	return dispatch => {
+		dispatch( recordTracksEvent( 'calypso_edit_gravatar_file_recieve_failure' ) );
+		dispatch( bumpStat( 'calypso_gravatar_update_error', statName ) );
+		dispatch( {
+			type: GRAVATAR_RECEIVE_IMAGE_FAILURE,
+			errorMessage,
+		} );
+	};
+}
+
+export function clickButton( { isVerified } ) {
+	return dispatch => {
+		dispatch( recordTracksEvent( 'calypso_edit_gravatar_click', { userVerified: isVerified } ) );
+		dispatch( recordGoogleEvent( 'Me', 'Clicked on Edit Gravatar Button in Profile' ) );
```

Comment:
Note that we have access to props in mapDispatchToProps:
```jsx
const mapDispatchToProps = ( dispatch, ownProps ) => {
    const isVerified = get( ownProps, 'user.isVerified' );
    //...
}; 
```

For a single stat bump the following is also an option:
```
const mapDispatchToProps = {
    uploadGravatarClick: partial( recordGoogleEvent, 'Me', 'Clicked on Edit Gravatar Button in Profile' );
};
```